.github/workflows/golang.yml

```diff
@@ -0,0 +1,22 @@
+name: Golang CI
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+jobs:
+
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up Go
+      uses: actions/setup-go@v3
+      with:
+        go-version: 1.18
+    - name: Build
+      run: go build -v ./...
```

Comment:
Should this also `go mod download` and/or do stuff with caching dependencies to speed up CI? I don't care if it's added now or if at all, but something to consider.